stack/app/agents/crag_message.py

```diff
@@ -0,0 +1,148 @@
+from typing import List, Optional, Union, Dict, Any
```

Comment:
This is not currently in use - it is here because it contains a potential solution for streaming the crag intermediate steps.